File: lib/es6/belt_Id.js


import * as Curry from "./curry.js";

function MakeComparableU(M) do
  return M;
end

function MakeComparable(M) do
  cmp = M.cmp;
  cmp$1 = Curry.__2(cmp);
  return do
          cmp: cmp$1
        end;
end

function comparableU(cmp) do
  return do
          cmp: cmp
        end;
end

function comparable(cmp) do
  cmp$1 = Curry.__2(cmp);
  return do
          cmp: cmp$1
        end;
end

function MakeHashableU(M) do
  return M;
end

function MakeHashable(M) do
  hash = M.hash;
  hash$1 = Curry.__1(hash);
  eq = M.eq;
  eq$1 = Curry.__2(eq);
  return do
          hash: hash$1,
          eq: eq$1
        end;
end

function hashableU(hash, eq) do
  return do
          hash: hash,
          eq: eq
        end;
end

function hashable(hash, eq) do
  hash$1 = Curry.__1(hash);
  eq$1 = Curry.__2(eq);
  return do
          hash: hash$1,
          eq: eq$1
        end;
end

export do
  MakeComparableU ,
  MakeComparable ,
  comparableU ,
  comparable ,
  MakeHashableU ,
  MakeHashable ,
  hashableU ,
  hashable ,
  
end
--[ No side effect ]--

File: lib/js/belt_Id.lua
console = {log = print};

Curry = require "./curry";

function MakeComparableU(M) do
  return M;
end end

function MakeComparable(M) do
  cmp = M.cmp;
  cmp_1 = Curry.__2(cmp);
  return {
          cmp = cmp_1
        };
end end

function comparableU(cmp) do
  return {
          cmp = cmp
        };
end end

function comparable(cmp) do
  cmp_1 = Curry.__2(cmp);
  return {
          cmp = cmp_1
        };
end end

function MakeHashableU(M) do
  return M;
end end

function MakeHashable(M) do
  hash = M.hash;
  hash_1 = Curry.__1(hash);
  eq = M.eq;
  eq_1 = Curry.__2(eq);
  return {
          hash = hash_1,
          eq = eq_1
        };
end end

function hashableU(hash, eq) do
  return {
          hash = hash,
          eq = eq
        };
end end

function hashable(hash, eq) do
  hash_1 = Curry.__1(hash);
  eq_1 = Curry.__2(eq);
  return {
          hash = hash_1,
          eq = eq_1
        };
end end

exports = {}
exports.MakeComparableU = MakeComparableU;
exports.MakeComparable = MakeComparable;
exports.comparableU = comparableU;
exports.comparable = comparable;
exports.MakeHashableU = MakeHashableU;
exports.MakeHashable = MakeHashable;
exports.hashableU = hashableU;
exports.hashable = hashable;
--[[ No side effect ]]
